### PR TITLE
Support built-in templates for Goodies

### PR DESCRIPTION
**Pull request by zachthompson**
Adds support for using the built-in Spice templates with Goodies.   Goodies can return a hash ref, e.g. [Goodie with built-in templating](https://github.com/zachthompson/zeroclickinfo-goodies/blob/goodie-template-v2/lib/DDG/Goodie/GoodieTemplateV2.pm), that essentially mimics the Spice.add call.

### CODE REVIEW

File: lib/App/DuckPAN/Web.pm [hunk 1]
```diff
@@ -285,11 +284,21 @@ sub request {
...
 					# There is no error-checking or support for non-auto-templates here.
 					my $structured = $result->structured_answer;
-					my $template_name = 'goodie_'.scalar @{$structured->{input}}.'_inputs';
-					my $json_string = encode_json({Answer => $structured});
-					$zci_container->push_content(HTML::TreeBuilder->new_from_content("<script>\$(window).load(function(){"
-								. "document.getElementById('zci-answer').innerHTML = DDG.exec_template('$template_name', $json_string);"
-								. "});</script>")->guts);
+					if(exists $structured->{templates}){ # user-specified templates
+						$page = $root->as_HTML;
+						$page =~ s|####DUCKDUCKHACK-CALL-NRJ####|;DDG.duckbar.future_signal_tab({signal:'high',from:'$structured->{id}'});|;
```
Comment (moollaza) [inline lib/App/DuckPAN/Web.pm]: We actually do this below starting at line 359. If you just create the script tag and it into the correct arrat, e.g. `@calls_nrj` this should work the same and then you don't have to both modifying the page here I believe?

Comment (moollaza) [inline lib/App/DuckPAN/Web.pm]: @zachthompson do you know if we plan to allow multiple Goodies now that they can have different tab names? Pushing into the `@calls_` arrays will also allow for that, if we want.

Comment (zachthompson) [inline lib/App/DuckPAN/Web.pm]: @moollaza I thought about taking that route but the needs of Goodies are a bit simpler and put together differently.  We would need to detect the difference and modify the construction below.  To me it seems like more work than is necessary.

We should try multiple goodies at some point but let's do it as a separate PR.

**Comment by moollaza** [comment]
Tested and works perfectly. LGTM :+1:

File: lib/App/DuckPAN/Web.pm [hunk 2]
```diff
@@ -285,11 +285,17 @@ sub request {
...
-					my $template_name = 'goodie_'.scalar @{$structured->{input}}.'_inputs';
-					my $json_string = encode_json({Answer => $structured});
-					$zci_container->push_content(HTML::TreeBuilder->new_from_content("<script>\$(window).load(function(){"
-								. "document.getElementById('zci-answer').innerHTML = DDG.exec_template('$template_name', $json_string);"
-								. "});</script>")->guts);
+					if(exists $structured->{templates}){ # user-specified templates
+						push @calls_goodie, $structured;
+						last;
+					}
+					else{ # auto-template
```
Comment (moollaza) [inline lib/App/DuckPAN/Web.pm]: @zachthompson aren't we're already using `DDG.duckbar()` to render the existing auto-templated Goodies?

Wondering if we can also use your `@calls_goodie` for these Goodie's as well? That way we can let the frontend deal with rendering rather than us shoving it into the page, which causes some problems (i.e #126)
